include/h2o/memory.h

```diff
@@ -54,7 +54,7 @@ extern "C" {
 #define H2O_GNUC_VERSION 0
 #endif
 
-#if __STDC_VERSION__ >= 201112L
+#if __STDC_VERSION__ >= 201112L && !defined(_MUSL_LIB)
```

Comment:
This is a C compiler issue rather than a libc issue. It has been resolved in #3427.